IdentityCore/src/cache/metadata/MSIDAccountMetadataCacheAccessor.m

```diff
@@ -77,25 +80,58 @@ - (BOOL)updateAuthorityURL:(NSURL *)cacheAuthorityURL
                    context:(id<MSIDRequestContext>)context
                      error:(NSError **)error
 {
+    if (!cacheAuthorityURL || !requestAuthorityURL || !homeAccountId || !clientId)
+    {
+        if (error) *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInvalidInternalParameter, @"Parameters cannot be nil for updating account metadata authority map!", nil, nil, nil, nil, nil, YES);
+        return NO;
+    }
+        
     //No need to update if the request authority is the same as the authority used internally
     if (!cacheAuthorityURL
         || [cacheAuthorityURL isEqual:requestAuthorityURL]) return YES;
     
-    MSIDAccountMetadataCacheKey *key = [[MSIDAccountMetadataCacheKey alloc] initWitHomeAccountId:homeAccountId clientId:clientId];
-    MSIDAccountMetadataCacheItem *accountMetadataItem = [_metadataCache accountMetadataWithKey:key context:context error:error];
-    if (!accountMetadataItem)
+    NSError *localError;
+    MSIDAccountMetadataCacheKey *key = [[MSIDAccountMetadataCacheKey alloc] initWithClientId:clientId];
+    MSIDAccountMetadataCacheItem *cacheItem = [_metadataCache accountMetadataCacheItemWithKey:key context:context error:&localError];
+    if (localError)
+    {
+        if (error) *error = localError;
+        return NO;
+    }
+    
+    if (!cacheItem)
+    {
+        cacheItem = [[MSIDAccountMetadataCacheItem alloc] initWithClientId:clientId];
+    }
+    
+    MSIDAccountMetadata *accountMetadata = [cacheItem accountMetadataForHomeAccountId:homeAccountId];
+    if (accountMetadata)
+    {
+        // No need to update if same record exists
+        if ([accountMetadata cachedURL:requestAuthorityURL instanceAware:instanceAware] == cacheAuthorityURL &&
```

Comment:
Should we do isEqual instead of "==" here?

Reply:
good catch! fixed :)